doc/build.md

```diff
@@ -0,0 +1,191 @@
+
+   Optional dependencies are automatically enabled if found by `cmake` on the
+   host build system. To disable their use, pass `-DUSE_<DEPENDENCY_NAME>=FALSE`
+   to `cmake` during configuration. Check out the [following section](#dependencies)
+   for more information about VCML's dependencies.
+
+   Release and debug build configurations are controlled via the regular
+   CMake parameters:
+   ```sh
+   cmake -B <build-dir> -DCMAKE_INSTALL_PREFIX=<install-dir> -DCMAKE_BUILD_TYPE=RELEASE <source-dir>
+   cmake --build <build-dir>
+   ```
+   If building with `-DVCML_BUILD_TESTS=ON`, you can run all unit tests using
+   `ctest` within `<build-dir>`.
+
+6. After building, VCML can then be installed into `<install-dir>` with the
+   following command:
+
+   ```sh
+   cmake --install <build-dir>
+   ```
+   Depending on the path, you may need elevated rights to write to the installation directory (e.g. with `sudo`).
+
+After installation, the following new files should be present:
+```
+<install-dir>/lib/libvcml.a   # library
+<install-dir>/include/vcml.h  # library header
+<install-dir>/include/vcml/   # header files
+<install-dir>/bin/            # utility programs/scripts
+```
+Update your environment so that other projects can reference your build:
+```sh
+export VCML_HOME=<install-dir>
+```
+
+## Windows Build & Installation
+
+Windows builds are currently supported using
+[Microsoft Visual Studio](https://visualstudio.microsoft.com/).
+There are two ways to build `vcml` on Windows:
+
+1. Using Visual Studio IDE:
+   - Launch Visual Studio
+   - In the Open Dialog click on `Clone a Repository`
+   - Enter `https://github.com/machineware-gmbh/vcml` and click `Clone`
+   - Once Visual Studio has cloned the project, double-click on the `vcml` folder
+   - Run `Build All` from the build menu.
+
+2. Using the command line:
+   - Install [Git for Windows](https://git-scm.com/download/win)
+   - Launch `git-bash` (installed by Git for Windows)
+   - Run `git clone --recursive https://github.com:machineware-gmbh/vcml`
+   - Run `cmake -B BUILD -G "Visual Studio 17" -DCMAKE_BUILD_TYPE=[DEBUG|RELEASE]`
+   - Run `cmake --build BUILD`
+
+## Maintaining Multiple Builds
+
+Debug builds (i.e. `-DCMAKE_BUILD_TYPE=DEBUG`) are intended for developers
+that use `vcml` to construct a new VP and want to track down bugs.
+Note that these builds operate significantly slower than optimized release
+builds and should therefore not be used for VPs that are used productively,
+e.g. for target software development. To maintain both builds from a single
+source repository, try the following:
+```sh
+git clone https://github.com/machineware-gmbh/vcml.git --recursive && cd vcml
+home=$PWD
+for type in "DEBUG" "RELEASE"; do
+    install="$home/BUILD/$type"
+    build="$home/BUILD/$type/BUILD"
+    mkdir -p $build && cd $build
+    cmake -DCMAKE_BUILD_TYPE=$type -DCMAKE_INSTALL_PREFIX=$install $home
+    cmake --build .
+    cmake --install .
+done
```

Comment:
This can also be achieved using the utility script at :
```
<source-dir>/utils/setup-gcc [DEBUG|RELEASE|...]
<source-dir>/utils/setup-clang [DEBUG|RELEASE|...]